src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/LibraryImportGenerator.cs

```diff
@@ -70,7 +71,20 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                         ? new { Syntax = (MethodDeclarationSyntax)context.TargetNode, Symbol = methodSymbol }
                         : null)
                 .Where(
-                    static modelData => modelData is not null);
+                    modelData =>
+                    {
+                        bool hasAttr = modelData is not null;
+                        anyLibraryImport |= hasAttr;
+                        return hasAttr;
+                    });
+
+            // If any methods are found to have the LibraryImportAttribute, we check
+            // to ensure necessary properties are set on the compilation context.
```

Comment:
I'd imagine it would have to restart anyways because the above is where we do the discovery of `LibraryImport` usage, but it is possible I'm not understanding this method's semantics. Let me investigate further.

Reply:
I see the issue. I misunderstood how this logic is used. I need to rework this approach.